### PR TITLE
Battery bq78z100

### PR DESCRIPTION
**Pull request by Qingwu-Li**
drivers:power:supply
--Add support for BQ27Z561, (cherry picked from commit 6f24ff9)
--Add support for BQ28z610, (cherry picked from commit 707d678)
--Add support for bq78z100 for Crocodile

Sorry for Closed the [PR7](https://github.com/hexagon-geo-surv/linux-leica/pull/7) and can not reopen.

### CODE REVIEW

**Comment by Qingwu-Li** [comment]
@zandrey @grembeter, Thanks a lot for input and knowlage share, I split into 3 comments, device tree and deconfig will checkin later, this PR just battery driver relevant.

**Comment by zandrey** [comment]
@Qingwu-Li Please see my comment in https://github.com/hexagon-geo-surv/linux-leica/pull/7#pullrequestreview-556249386 regarding the commit you've introduced - it should be prepared to be upstreamed.

**Comment by Qingwu-Li** [comment]
@zandrey, OK, I'm working on prepare the patchs.

File: drivers/power/supply/bq27xxx_battery.c [hunk 1]
```diff
@@ -43,6 +43,7 @@
  * https://www.ti.com/product/bq27621-g1
  * https://www.ti.com/product/bq27z561
  * https://www.ti.com/product/bq28z610
+ * https://www.ti.com/product/BQ78Z100
```
Comment (zandrey) [inline drivers/power/supply/bq27xxx_battery.c:46]: Please use all small letters here (same as for all links above).

**Review by zandrey — CHANGES_REQUESTED** [review]
There are style problems with commit a97f194df44b2a34c56056b60869320b764c8a5b, please run `./scripts/checkpatch.pl` and have them fixed.

In addition, as the new `ti,bq78z100` is introduced - a separate documentation patch would be required here, see [Documentation/devicetree/bindings/submitting-patches.txt](../../../blob/stable/linux-5.4.y/Documentation/devicetree/bindings/submitting-patches.txt)

**Comment by Qingwu-Li** [comment]
@zandrey @grembeter This patch can not upstream to master, since there are big difference.  I prepared another branch, it can be upstream to master, but can not build in Crocodile, due to the head file of 'include/linux/device.h', (new function dev_err_probe) the change and also introduced by many drivers, an also the documents changed to yaml in the master.
[The link of another branch picked from master.](https://github.com/hexagon-geo-surv/linux-leica/compare/hexagon-geo-surv:leica/5.4-2.1.x-imx/crocodile...Qingwu-Li:power_supply_master)

**Comment by Qingwu-Li** [comment]
This PR expired, the change was upsteam to master.

**Pull request closed without merging**